src/uucore/src/lib/mods/locale.rs

```diff
@@ -0,0 +1,330 @@
+        let paths_str = tried_paths
+            .iter()
+            .map(|p| p.to_string_lossy().to_string())
+            .collect::<Vec<_>>()
+            .join(", ");
+
+        return Err(LocalizationError::Io {
+            source: std::io::Error::new(
+                std::io::ErrorKind::NotFound,
+                "No localization files found",
+            ),
+            path: PathBuf::from(paths_str),
+        });
+    }
+
+    Ok(bundle)
+}
+
+fn get_message_internal(id: &str, args: Option<FluentArgs>, default: &str) -> String {
+    LOCALIZER.with(|lock| {
+        lock.get()
+            .map(|loc| loc.format(id, args.as_ref(), default))
+            .unwrap_or_else(|| default.to_string())
+    })
+}
+
+/// Retrieves a localized message by its identifier.
+///
+/// Looks up a message with the given ID in the current locale bundle and returns
+/// the localized text. If the message ID is not found, returns the provided default text.
+///
+/// # Arguments
+///
+/// * `id` - The message identifier in the Fluent resources
+/// * `default` - Default text to use if the message ID isn't found
+///
+/// # Returns
+///
+/// A `String` containing either the localized message or the default text
+///
+/// # Examples
+///
+/// ```
+/// use uucore::locale::get_message;
+///
+/// // Get a localized greeting or fall back to English
+/// let greeting = get_message("greeting", "Hello, World!");
+/// println!("{}", greeting);
+/// ```
+pub fn get_message(id: &str, default: &str) -> String {
+    get_message_internal(id, None, default)
+}
+
+/// Retrieves a localized message with variable substitution.
+///
+/// Looks up a message with the given ID in the current locale bundle,
+/// substitutes variables from the provided arguments map, and returns the
+/// localized text. If the message ID is not found, returns the provided default text.
+///
+/// # Arguments
+///
+/// * `id` - The message identifier in the Fluent resources
+/// * `rustc_args` - Key-value pairs for variable substitution in the message
+/// * `default` - Default text to use if the message ID isn't found
+///
+/// # Returns
+///
+/// A `String` containing either the localized message with variable substitution or the default text
+///
+/// # Examples
+///
+/// ```
+/// use uucore::locale::get_message_with_args;
+/// use std::collections::HashMap;
+///
+/// // For a Fluent message like: "Hello, { $name }! You have { $count } notifications."
+/// let mut args = HashMap::new();
+/// args.insert("name".to_string(), "Alice".to_string());
+/// args.insert("count".to_string(), "3".to_string());
+///
+/// let message = get_message_with_args(
+///     "notification",
+///     args,
+///     "Hello! You have notifications."
+/// );
+/// println!("{}", message);
+/// ```
+pub fn get_message_with_args(
+    id: &str,
+    rustc_args: HashMap<String, String>,
+    default: &str,
+) -> String {
+    let mut args = FluentArgs::new();
+    for (k, v) in rustc_args {
+        args.set(k, v);
+    }
```

Comment:
I think you can simplify it to:
```suggestion
    let args = rustc_args.into_iter().collect();
```